ts/src/index.ts

```diff
@@ -116,25 +123,21 @@ const profiler: Profiler|undefined = undefined;
 
 /**
  * Initializes the config, and starts heap profiler if the heap profiler is
- * needed. Returns a profiler if creation is successful. Otherwise, returns
- * undefined.
+ * needed. Returns a profiler if creation is successful. Otherwise, throws an
+ * error.
  */
-export async function createProfiler(config: Config):
-    Promise<Profiler|undefined> {
+export async function createProfiler(config: Config): Promise<Profiler> {
   if (!semver.satisfies(process.version, pjson.engines.node)) {
-    logError(
+    throw new Error(
         `Could not start profiler: node version ${process.version}` +
-            ` does not satisfies "${pjson.engines.node}"`,
-        config);
-    return undefined;
+        ` does not satisfies "${pjson.engines.node}"`);
   }
 
   let profilerConfig: ProfilerConfig;
   try {
     profilerConfig = initConfigLocal(config);
   } catch (e) {
-    logError(`Could not start profiler: ${e}`, config);
-    return undefined;
+    throw new Error(`Could not start profiler: ${e}`);
```

Comment:
It would be better to throw the original error. The additional message doesn't really add value. That is, I think you can omit the try/catch.

Reply:
Done.